prefect_dbt/cloud/clients.py

```diff
@@ -29,6 +29,34 @@ def __init__(self, api_key: str, account_id: int, domain: str):
             base_url=f"https://{domain}/api/v2/accounts/{account_id}",
         )
 
+    async def call_endpoint(
+        self,
+        http_method: str,
+        path: str,
+        params: Optional[Dict[str, Any]] = None,
+        json: Optional[Dict[str, Any]] = None,
+    ) -> Response:
+        """
+        Call an endpoint in the dbt Cloud API.
+
+        Args:
+            path: The partial path for the request (e.g. /projects/). Will be appended
+                onto the base URL as determined by the client configuration.
+            http_method: HTTP method to call on the endpoint.
+            params: Query parameters to include in the request.
+            json: JSON serializable body to send in the request.
+
+        Returns:
+            The response from the dbt Cloud administrative API.
+        """
+        response = await self._admin_client.request(
+            method=http_method, url=path, params=params, json=json
```

Comment:
Nice! wasn't aware of the request method 👍

Reply:
Should other methods call this method, like `trigger_job_run` to remove redundant `response.raise_for_status()`

Reply:
Good call! That cleaned up the other client methods nicely.